src/super_gradients/training/utils/bbox_formats/bbox_format.py

```diff
@@ -0,0 +1,118 @@
+        scale = np.array([cols, rows, cols, rows], dtype=bboxes.dtype)
+    else:
+        raise RuntimeError(f"Only Torch tensor or Numpy array is supported. Received bboxes of type {str(type(bboxes))}")
+    return bboxes / scale
+
+
+def normalizedxyxy2xyxy_inplace(bboxes: Union[Tensor, np.ndarray], image_shape: Tuple[int, int]) -> Union[Tensor, np.ndarray]:
+    """
+    Convert unit-normalized XYXY bboxes to XYXY bboxes in pixel units. This function operates in-place.
+    :param bboxes: BBoxes of shape (..., 4) in XYXY (unit-normalized) format
+    :param image_shape: Image shape (rows,cols)
+    :return: BBoxes of shape (..., 4) in XYXY (pixels) format
+    """
+    rows, cols = image_shape
+    bboxes[..., 0:3:2] *= cols
+    bboxes[..., 1:4:2] *= rows
+    return bboxes
+
+
+def xyxy2normalizedxyxy_inplace(bboxes: Tensor, image_shape: Tuple[int, int]) -> Tensor:
+    """
+    Convert bboxes from XYXY (pixels) format to XYXY (unit-normalized) format. This function operates in-place.
+    :param bboxes: BBoxes of shape (..., 4) in XYXY (pixels) format
+    :param image_shape: Image shape (rows,cols)
+    :return: BBoxes of shape (..., 4) in XYXY (unit-normalized) format
+    """
+    rows, cols = image_shape
+    bboxes[..., 0:3:2] /= cols
+    bboxes[..., 1:4:2] /= rows
+    return bboxes
+
+
+class BoundingBoxFormat:
+    """
+    Abstract class for describing a bounding boxes format. It exposes two methods: to_xyxy and from_xyxy to convert
+    whatever format of boxes we are dealing with to internal xyxy format and vice versa. This conversion from and to
+    intermediate xyxy format has a subtle performance impact, but greatly reduce amount of boilerplate code to support
+    all combinations of conversion xyxy, xywh, cxcywh, yxyx <-> xyxy, xywh, cxcywh, yxyx.
+    """
+
+    @abstractmethod
+    def to_xyxy(self, bboxes: Union[Tensor, np.ndarray], image_shape: Tuple[int, int], inplace: bool) -> Union[Tensor, np.ndarray]:
+        """
+        Convert input boxes to XYXY format
+        :param bboxes: Input bounding boxes [..., 4]
+        :param image_shape: Dimensions (rows, cols) of the original image to support
+                            normalized boxes or non top-left origin coordinate system.
+        :return: Converted bounding boxes [..., 4] in XYXY format
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def from_xyxy(self, bboxes: Union[Tensor, np.ndarray], image_shape: Tuple[int, int], inplace: bool) -> Union[Tensor, np.ndarray]:
+        """
+        Convert XYXY boxes to target bboxes format
+        :param bboxes: Input bounding boxes [..., 4] in XYXY format
+        :param image_shape: Dimensions (rows, cols) of the original image to support
+                            normalized boxes or non top-left origin coordinate system.
+        :return: Converted bounding boxes [..., 4] in target format
+        """
+        raise NotImplementedError()
+
+
+def convert_bboxes(
```

Comment:
Is there a reason why you defined convert_bboxes as a function and not a method ?

It would look like this:
`def to(self, bboxes, image_shape, target_format):
    xyxy = self.to_xyxy(bboxes, image_shape, inplace)
    return target_format.from_xyxy(xyxy, image_shape, inplace)`